repobee_sanitizer/_sanitize_repo.py

```diff
@@ -102,6 +107,44 @@ def create_extension_command(self) -> plug.ExtensionCommand:
         )
 
 
+def _parse_file_list(file_list) -> List[pathlib.Path]:
+    return [
+        p.strip()
+        for p in file_list.read_text(encoding=_ASSUMED_ENCODING)
+        .strip()
+        .split("\n")
+    ]
+
+
+def _discover_dirty_files(repo_root) -> List[pathlib.Path]:
+    """
+    Returns:
+        A list of relative file paths for files that need to be sanitized.
+    """
+    return [
+        file
+        for file in list(repo_root.rglob("*"))
+        if ".git" not in str(file) and file.is_file() and _file_is_dirty(file)
+    ]
```

Comment:
This is not quite correct, and it's a good example of why we should try to avoid using paths as strings: it's easy to mess up. For example, if a file (for some reason) is called `BestGitTricks.git`, then that will not be found. Anything in e.g. the `.github` directory will also be ignored. What you want to do here is to specifically ignore files that are in the `.git` directory, or any of its subdirectories.

I could tell you how to correct this, but I think it's better if you try on your own. Here's a start.

```suggestion
def _discover_dirty_files(repo_root) -> List[pathlib.Path]:
    """
    Returns:
        A list of relative file paths for files that need to be sanitized.
    """
    git_dir = repo_root / ".git"
    assert git_dir.is_dir()
    return [
        file
        for file in list(repo_root.rglob("*"))
        if # file is not in git_dir or any of its subdirs
    ]
```